packages/api/src/controllers/stream.ts

```diff
@@ -389,23 +414,25 @@ export function getRecordingFields(
   const isUnused = !session.lastSeen && session.createdAt < readyThreshold;
 
   const recordingStatus = isReady ? "ready" : isUnused ? "none" : "waiting";
-  return !isReady && !forceUrl
-    ? { recordingStatus }
-    : {
-        recordingStatus,
-        recordingUrl: getRecordingUrl(ingest, session),
-        mp4Url: getRecordingUrl(ingest, session, true),
-      };
+  if (!isReady && !forceUrl) {
+    return { recordingStatus };
+  }
+  const recordingUrl = await getRecordingUrl(ingest, session);
+  const mp4Url = await getRecordingUrl(ingest, session, true);
```

Comment:
Could we get both fields in a single call? Otherwise, we'll be going 4 times to the DB to fetch the OS and the asset twice each.

Reply:
Updated in a10bb97278aa3e333629469accb329d4b888cf5f